### DISCUSSION TITLE
Premières analyses des DUMPS - Organizer key "@id"

### DISCUSSION THREAD

**Discussion by GenFab28** [culturecreates/artsdata-planet-lavitrine#26] [Modelling discussions]
En important les DUMPS dans le système de La Vitrine, voici les observations sur certaines inconsistances / à discuter voir si on peut améliorer les importations

We found a difference in the format of some data using artsdata API : 
For some organizer key the “@id” key is missing, when this field is missing a i am currently using the url key as id.
It would be cleaner and simpler to be able to use the @id key for all organizer. 
Exemple of ‘classic’ data :

"organizer": [
        {
          "@id": "https://www.grandtheatre.qc.ca/programmation/le-theatre-du-trident",
          "type": "Organization",
          "sameAs": [
            "https://fr.wikipedia.org/wiki/Th%C3%A9%C3%A2tre_du_Trident",
            "https://www.letrident.com/",
            "https://www.facebook.com/Th%C3%A9%C3%A2tre-du-Trident-126893944388/"
          ],
          "name": {
            "@none": "Le Trident"
          },
          "url": "https://www.letrident.com/"
        }
      ],

**And from the same Json file** 

"organizer": [
        {
          "type": "Organization",
          "sameAs": [
            "https://www.instagram.com/grandtheatreqc/",
            "https://twitter.com/GrandTheatreQc",
            "http://www.wikidata.org/entity/Q3114610",
            "https://fr.wikipedia.org/wiki/Grand_Th%C3%A9%C3%A2tre_de_Qu%C3%A9bec",
            "https://www.facebook.com/grandtheatre",
            "https://www.youtube.com/channel/UCcrgPnIkEmTMe7tvJRiUMow"
          ],
          "name": {
            "@none": "Grand Théâtre de Québec"
          },
          "url": "https://grandtheatre.qc.ca/"
        }
    ],

@saumier @christianroy

**Comment by saumier** [comment]
We should analyze this further because for GTQ we are mostly using their structured data so in this case it is GTQ that is not using @id.  My first impression is that it would be much much more stable to use the Artsdata persistent identifier. As in our last call, I think that I should be adding these in when they are missing. Lets discuss at our Wednesday meeting.

**Reply by GenFab28** [reply]
Perfect

**Reply by christianroy** [reply]
Je suis retombé sur ce thread suite à une question de l'équipe de La Vitrine en l'absence de Geneviève qui tente de comprendre pourquoi certaines contributions sont mal importées de leur côté. J'ai du mal à identifier clairement le problème (ils ont corrigé manuellement donc je n'ai plus accès aux valeurs originales), mais je suspecte que ce sont seulement les liens de contribution associés à l'organisateur pour le Grand théâtre...

Je suis en train de vérifier avec eux si l'affirmation initiale «when this field is missing a i am currently using the url key as id» est vraie. Si c'est le cas, j'ai probablement mal identifié la source du problème. Sinon, le problème est probablement celui décrit dans cette discussion. 

J'avoue ne plus être certain si le sujet a été abordé lors du Wednesday meeting évoqué... 😅 Mais est-ce que l'ajout d'un @id persistent Artsdata lorsque rien n'est spécifié est fait?
